lib/llvmopencl/Workgroup.cc

```diff
@@ -267,6 +267,16 @@ bool WorkgroupImpl::runOnModule(Module &M, llvm::FunctionAnalysisManager &FAM) {
   // extra printf arguments.
   FunctionMapping PrintfCache;
 
+  // Remove the OptNone&NoInline keywords from all functions;
+  for (Module::iterator It = M.begin(), e = M.end(); It != e; ++It) {
+    Function &F = *It;
```

Comment:
I think, this simplifies to:
```suggestion
  for (Function &F : M) {
```